core/shared/src/main/scala/zio/Cause.scala

```diff
@@ -535,10 +605,16 @@ object Cause extends Serializable {
   private[Cause] def sym(f: (Cause[Any], Cause[Any]) => Boolean): (Cause[Any], Cause[Any]) => Boolean =
     (l, r) => f(l, r) || f(r, l)
 
-  private[Cause] def flatten(c: Cause[Any]): Set[Cause[Any]] = c match {
-    case Then(left, right) => flatten(left) ++ flatten(right)
-    case Both(left, right) => flatten(left) ++ flatten(right)
-    case Traced(cause, _)  => flatten(cause)
+  private[Cause] def flattenSeq(c: Cause[_]): Vector[Cause[_]] = c match {
+    case Then(left, right) => flattenSeq(left) ++ flattenSeq(right)
```

Comment:
Missing Both?

Reply:
`flattenSeq` should not flatten both. @adamgfraser fixed all this logic in the new one.